reframe/listener.go

```diff
@@ -0,0 +1,564 @@
+package reframelistener
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"math/rand"
+	"sort"
+	"time"
+
+	provider "github.com/filecoin-project/index-provider"
+	"github.com/filecoin-project/index-provider/metadata"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	dsq "github.com/ipfs/go-datastore/query"
+	"github.com/ipfs/go-delegated-routing/client"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-multihash"
+)
+
+var log = logging.Logger("reframe/listener")
+var bitswapMetadata = metadata.New(metadata.Bitswap{})
+
+const (
+	chunkByContextIdIndexPrefix = "reframe/ccid/"
+	timestampByCidIndexPrefix   = "reframe/tc/"
+)
+
+type ReframeListener struct {
+	ds           datastore.Datastore
+	e            provider.Interface
```

Comment:
e is a bit too short, i think this is standing for `engine`?

Reply:
indeed - renamed